### ISSUE TITLE
Unable to access AKS Clusters

### ISSUE THREAD

**Issue by cpressland** [kr8s-org/kr8s#161]
### Which project are you reporting a bug for?

kr8s

### What happened?

After installing on Python 3.11.5, and attempting to use the example in the documentation:

```python
import kr8s

pods = kr8s.get("pods")
```

I get the following exception:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/_io.py", line 46, in wrapped
    return portal.call(wrapped)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/anyio/from_thread.py", line 261, in call
    return cast(T_Retval, self.start_task_soon(func, *args).result())
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cpressland/.pyenv/versions/3.11.5/lib/python3.11/concurrent/futures/_base.py", line 456, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/cpressland/.pyenv/versions/3.11.5/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/anyio/from_thread.py", line 198, in _call_func
    retval = await retval
             ^^^^^^^^^^^^
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/asyncio/_helpers.py", line 22, in get
    api = await _api(_asyncio=_asyncio)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/asyncio/_api.py", line 44, in api
    return await _f(
           ^^^^^^^^^
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/asyncio/_api.py", line 42, in _f
    return await _cls(**kwargs, bypass_factory=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/_api.py", line 59, in f
    await self.auth
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/_auth.py", line 48, in f
    await self.reauthenticate()
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/_auth.py", line 59, in reauthenticate
    await self._load_kubeconfig()
  File "/Users/cpressland/dev/starbug/.venv/lib/python3.11/site-packages/kr8s/_auth.py", line 122, in _load_kubeconfig
    **{e["name"]: e["value"] for e in self._user["exec"].get("env", [])}
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```

The issue _might_ be coming from the fact that kubectl auth is provided by [kubelogin](https://github.com/Azure/kubelogin).

### Anything else?

Using the `kubectl proxy` solution documented [here](https://docs.kr8s.org/en/latest/authentication.html#url) is a semi functional solution as follow up commands return a number of other exceptions:

```python
>>> import kr8s
>>> client = kr8s.api(url="http://127.0.0.1:8000")
>>> pods = kr8s.get("pods")
>>> pods[0].get()
TypeError: 'NoneType' object is not iterable
>>> pods[0].get()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)
```

**Comment by jacobtomlinson** [comment]
I think this should be fixed by #156

**Comment by cpressland** [comment]
Thanks @jacobtomlinson, tested via `$ poetry add git+ssh://git@github.com:kr8s-org/kr8s.git` and all working as expected. Looking forward to the next release.
